### PR TITLE
Added guidance for repeatable requests

### PR DESCRIPTION
**Pull request by johanste**

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

File: azure/Guidelines.md [hunk 1]
```diff
@@ -296,6 +296,16 @@ Clients that use version discovery are expected to cache version information. Si
...
 
+## Repeatability of requests
+
+The ability to retry failed requests for which a client never received a response greatly simplifies the ability to write resilient distributed applications. While HTTP designates some methods as safe and/or idempotent (and thus retryable), being able to retry other operations such as create-using-POST-to-collection is desirable.
+
+A service **SHOULD** support repeatable requests according as defined in [OASIS Repeatable Requests Version 1.0](https://docs.oasis-open.org/odata/repeatable-requests/v1.0/repeatable-requests-v1.0.html).
+
+- The tracked time window (difference between the `Repeatability-First-Sent` value and the current time) **MUST** be at least 5 minutes.
+- A service advertises support for repeatability requests by adding the `Repeatbility-First-Sent` and `Repeatability-Request-ID` to the set of headers for a given operation.
```
Comment (bojunehsu) [inline azure/Guidelines.md:306]: Typo: Repeat**a**bility-First-Sent

File: azure/Guidelines.md [hunk 2]
```diff
@@ -296,6 +296,16 @@ Clients that use version discovery are expected to cache version information. Si
...
 
+## Repeatability of requests
+
+The ability to retry failed requests for which a client never received a response greatly simplifies the ability to write resilient distributed applications. While HTTP designates some methods as safe and/or idempotent (and thus retryable), being able to retry other operations such as create-using-POST-to-collection is desirable.
+
+A service **SHOULD** support repeatable requests according as defined in [OASIS Repeatable Requests Version 1.0](https://docs.oasis-open.org/odata/repeatable-requests/v1.0/repeatable-requests-v1.0.html).
```
Comment (mkistler) [inline azure/Guidelines.md]:
```suggestion
A service **SHOULD** support repeatable requests as defined in [OASIS Repeatable Requests Version 1.0](https://docs.oasis-open.org/odata/repeatable-requests/v1.0/repeatable-requests-v1.0.html).
```

File: azure/Guidelines.md [hunk 3]
```diff
@@ -296,6 +296,16 @@ Clients that use version discovery are expected to cache version information. Si
...
+
+- The tracked time window (difference between the `Repeatability-First-Sent` value and the current time) **MUST** be at least 5 minutes.
+- A service advertises support for repeatability requests by adding the `Repeatbility-First-Sent` and `Repeatability-Request-ID` to the set of headers for a given operation.
+- When understood, all endpoints co-located behind a DNS name **MUST** understand the header. This means that a service **MUST NOT** ignore the presence of a header for any endpoints behind the DNS name, but rather fail the request containing a `Repeatability-Request-ID` header if that particular endpoint lacks support for repeatable requests. Such partial support **SHOULD** be avoided due to the confusion it causes for clients.
```
Comment (mkistler) [inline azure/Guidelines.md]:
```suggestion
- When supported, all endpoints co-located behind a DNS name **MUST** support the header. This means that a service **MUST NOT** ignore the presence of a header for any endpoints behind the DNS name, but rather fail the request containing a `Repeatability-Request-ID` header if that particular endpoint lacks support for repeatable requests. Such partial support **SHOULD** be avoided due to the confusion it causes for clients.
```

**Review by mkistler — APPROVED** [review]
Looks good.  I made a few minor edits that I hope you accept but aren't essential.

**Pull request merged**